sg_AbstractSpatAlgorithm.cpp

```diff
@@ -31,6 +31,89 @@
+
+    void initialise() override
+    {
+        // init souce buffer
+        juce::Array<source_index_t> sources;
+        for (int i = 1; i <= MAX_NUM_SOURCES; ++i)
+            sources.add(source_index_t{ i });
+        sourceBuffer.init(sources);
+
+        // init speaker buffer
+        juce::Array<output_patch_t> speakers;
+        for (int i = 1; i <= MAX_NUM_SPEAKERS; ++i)
+            speakers.add(output_patch_t{ i });
+        speakerBuffer.init(speakers);
+
+        isRunning = true;
+    }
+
+    void runTest() override
+    {
+        beginTest("VBAP test");
+        {
+            // init project data and audio config
+            SpatGrisData vbapData;
+            vbapData.speakerSetup = *SpeakerSetup::fromXml(*parseXML(DEFAULT_SPEAKER_SETUP_FILE));
+            vbapData.project = *ProjectData::fromXml(*parseXML(DEFAULT_PROJECT_FILE));
+            vbapData.project.spatMode = SpatMode::vbap;
+            vbapData.appData.stereoMode = {};
+            auto const vbapConfig{ vbapData.toAudioConfig() };
+
+            auto vbapAlgorithm{ AbstractSpatAlgorithm::make(vbapData.speakerSetup,
+                                                            vbapData.project.spatMode,
+                                                            vbapData.appData.stereoMode,
+                                                            vbapData.project.sources,
+                                                            vbapData.appData.audioSettings.sampleRate,
+                                                            vbapData.appData.audioSettings.bufferSize) };
+
+            for (int i = 0; i < numLoops; ++i)
+                vbapAlgorithm->process(*vbapConfig, sourceBuffer, speakerBuffer, stereoBuffer, sourcePeaks, nullptr);
+        }
+
+        beginTest("HRTF test");
+        {
+            SpatGrisData hrtfData;
+
+            hrtfData.speakerSetup = *SpeakerSetup::fromXml(*parseXML(BINAURAL_SPEAKER_SETUP_FILE));
+            hrtfData.project = *ProjectData::fromXml(*parseXML(DEFAULT_PROJECT_FILE));
+            hrtfData.project.spatMode = SpatMode::vbap;
+            hrtfData.appData.stereoMode = StereoMode::hrtf;
+            auto const hrtfConfig{ hrtfData.toAudioConfig() };
+
+            auto hrtfAlgorithm{ AbstractSpatAlgorithm::make(hrtfData.speakerSetup,
+                                                            hrtfData.project.spatMode,
+                                                            hrtfData.appData.stereoMode,
+                                                            hrtfData.project.sources,
+                                                            hrtfData.appData.audioSettings.sampleRate,
+                                                            hrtfData.appData.audioSettings.bufferSize) };
+
+            for (int i = 0; i < numLoops; ++i)
+                hrtfAlgorithm->process(*hrtfConfig, sourceBuffer, speakerBuffer, stereoBuffer, sourcePeaks, nullptr);
+        }
```

Comment:
do we want to have some assertions in these tests ? e.g. is there anything useful we could test besides "not crashing" ? for audio in general I try to at least have some checks to make sure that we don't get NaNs or other weird values in our output buffers ; also, it would make sense to try with varied buffer sizes e.g. from 1 to 8192